examples/se2_inverse.py

```diff
@@ -0,0 +1,65 @@
+# This example will use torch to solve min_x1 || x1^-1 - x2 ||^2,
+# where x1 and x2 are SE2 objects
+
+import torch
+import theseus as th
+from theseus import LieGroupTensor
+from theseus.geometry.lie_group import LieGroup
+
+from theseus.geometry.tests.test_se2 import create_random_se2
+
+rng = torch.Generator()
+rng.manual_seed(0)
+x1 = create_random_se2(1, rng)
+x2 = create_random_se2(1, rng)
+
+
+def run(x1: LieGroup, x2: LieGroup, num_iters=10, use_lie_tangent=True):
+    if use_lie_tangent:
+        # update x1 with the Lie group tangent gradient
+        x1.data.__class__ = LieGroupTensor
+        x1.data.__dict__["group_cls"] = type(x1)
+    else:
+        # update x1 with the Euclidean gradient
+        x1.data.__class__ = torch.Tensor
+        x1.data.__dict__.pop("group_cls", None)
```

Comment:
I think if we can find a way to add this logic to the LieGroup class itself and use a variable (like dtype or device) to specific which grad mode to use then the overall api would be a lot more user friendly.

Reply:
> I think if we can find a way to add this logic to the LieGroup class itself and use a variable (like dtype or device) to specific which grad mode to use then the overall api would be a lot more user friendly.

I'm OK for either.